jetty-core/jetty-server/src/main/java/org/eclipse/jetty/server/internal/HttpConnection.java

```diff
@@ -108,7 +108,7 @@ public class HttpConnection extends AbstractConnection implements Runnable, Writ
     private final LongAdder bytesOut = new LongAdder();
     private final AtomicBoolean _handling = new AtomicBoolean(false);
     private final HttpFields.Mutable _headerBuilder = HttpFields.build();
-    private volatile RetainableByteBuffer _retainableByteBuffer;
+    private final AtomicReference<RetainableByteBuffer> _retainableByteBuffer = new AtomicReference<>();
```

Comment:
Why was it necessary to use an atomic here?
As this field is "hot", so avoiding an indirection would be good.

However, the changes to atomic revealed that the code before was not really paying attention to the fact that the field was `volatile`, so the code is now better, but with an indirection.
Can we have better without indirection?